### ISSUE TITLE
Audio compressor do not work ?

### ISSUE THREAD

**Issue by Npa972** [quodlibet/quodlibet#4456]
Steps to reproduce
------------------
I tried to use audio compressor on QL but i do not hear any difference compare to Audacious audio compressor on same track.

I tried different settings without hearing any audio changes.

Did i miss something ?

Thanks.

Expected Output
---------------
For example, with a classical track starting very low, the volume should be able to be increased with audio compression.
This works perfectly with audacious.

Actual Output
-------------
I don't hear any change using audio compression.

Test System
-----------

### Which version of Quod Libet?
Quodlibet-git archlinux

### Which operating system 
archlinux

### If it's audio-related, what back-end?
pipewire

**Comment by declension** [comment]
Yes, I can replicate this.

Though I've never actually used it (and maybe I'm missing something) but I don't see how the plugin would have actually worked as the ratio (currently) only goes up to 1 which seems pointless to me. Also, a quick scan of the [GST source code](https://github.com/GStreamer/gst-plugins-good/blob/master/gst/audiofx/audiodynamic.c) makes me think it's actually acting on sample values, not time-smoothed RMS volume (or better loudness) which is... bad. But hey 

Personally I'd suggest running the QL output through a proper compressor - if you're on pipewire then you can use Jack output and all the Jack options, e.g. [Calf](https://calf-studio-gear.org/)

**Comment by Npa972** [comment]
Yes you are missing something especially with classical music (for my taste...).
I tried easyeffects but it's a big machine and I would prefer to use the plugins integrated into QL.
I know a bit about calf (which is needed for easyeffects) and I might try it.
But why keep a plugin that doesn’t work or works poorly?
The QL equalizer is very good but I'm missing a crystalizer plugin, an extra stereo plugin (and a functional audio compressor plugin).
I hope one day these plugins will be integrated into QL.
Anyway thank you for your reply !

**Comment by Npa972** [comment]
@declension Hi, is it possible to contact the author of this plugin to ask for a fix ?
Thanks.

**Comment by declension** [comment]
> @declension Hi, is it possible to contact the author of this plugin to ask for a fix ? Thanks.

Well I see https://github.com/GStreamer/gst-plugins-good/blob/20bbeb5e37666c53c254c7b08470ad8a00d97630/gst/audiofx/audiodynamic.c#L3 but doubt you'd have much luck there given how old it is. I think a new plugin would be more realistic.
